airbyte-webapp/src/components/connectorBuilder/StreamTestingPanel/PageDisplay.tsx

```diff
@@ -40,7 +40,16 @@ export const PageDisplay: React.FC<PageDisplayProps> = ({ page, className, infer
 
   const formattedRecords = useMemo(() => formatJson(page.records), [page.records]);
   const formattedRequest = useMemo(() => formatJson(page.request), [page.request]);
-  const formattedResponse = useMemo(() => formatJson(page.response), [page.response]);
+  const formattedResponse = useMemo(() => {
+    if (!page.response) {
+      return "";
+    }
+    const unpackedBodyResponse = {
+      ...page.response,
+      body: page.response.body ? JSON.parse(page.response.body) : undefined,
```

Comment:
as mentioned in the other comment, the connector builder server could return a plain string if the calling API allows for it. Can we use a conditional block to test the body string is valid json first, or try/catch on the json parse. Whichever is idiomatic for typescript

Reply:
If the contract is “this is the body, it might be json, it might be not”, I agree with keeping it named body - I will add the try/catch on Monday to be future-safe